zipline/_protocol.pyx

```diff
@@ -21,6 +21,28 @@ from cpython cimport bool
 
 from zipline.assets import Asset
 
+
+class assert_keywords(object):
+    """
+    Asserts that the keywords passed into the wrapped function are included
+    in those passed into this decorator. If not, raise a TypeError with a
+    meaningful message, unlike the one cython returns by default.
+    """
+
+    def __init__(self, *args):
+        self.names = args
+
+    def __call__(self, func):
+        def assert_keywords_and_call(*args, **kwargs):
+            for field in kwargs.keys():
```

Comment:
Calling `keys()` is redundant, iterating over a dict `for field in kwargs` has the same behavior.